pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/mutable/provider/MutableIndexContext.java

```diff
@@ -18,36 +18,97 @@
  */
 package org.apache.pinot.segment.spi.index.mutable.provider;
 
+import java.io.File;
 import java.util.Objects;
+import javax.annotation.Nullable;
 import org.apache.pinot.segment.spi.memory.PinotDataBufferMemoryManager;
-import org.apache.pinot.spi.config.table.JsonIndexConfig;
 import org.apache.pinot.spi.data.FieldSpec;
 
 
-public interface MutableIndexContext {
-  PinotDataBufferMemoryManager getMemoryManager();
+public class MutableIndexContext {
```

Comment:
Like we did in IndexCreationContext during `index-spi`, the idea here is:
- Make the interface a class
- Move the logic from Common to the class
- Remove Wrapper and all extra flavors.

Instead of using the extra flavors, each index type receives this context and their own configuration.

I also added 3 extra attributes to the class:
- _estimatedColSize
- _estimatedCardinality
- _avgNumMultiValues

As they were used by some indexes and may be useful for future indexes.

Reply:
I think I forgot to push something. It should be fixed now

Reply:
pardon me, what's the new fixes you pushed? got a link to them?

Reply:
Oh, sorry, I thought you mean to change MutableIndexContext from interface to class. I didn't touch `IndexCreationContext`. In case we change it, it should be done in another PR